### PR TITLE
Feat: add Unit Configuration Modal

### PR DESCRIPTION
**Pull request by CefBoud**
**Description:** 

This PR adds Unit Configuration modal and some minor refactoring.

**Depends on**: 

* https://github.com/openedx/frontend-app-course-authoring/pull/776

**Testing instructions**:

* Setup this repo as described in the readme.
* Checkout branch from https://github.com/openedx/edx-platform/pull/33667 and run `make cms-up` in devstack.
* Click on a Unit's top right ellipsis menu then open the Configuration Modal
* Test different changes to the Unit configuration. 

Bonus: create cohorts and restrict unit access to certain cohorts/content groups.

**Screenshot:**
Screenshot from 2024-01-11 17-32-04

Screenshot from 2024-01-11 17-30-59

Screenshot from 2024-01-11 17-31-39

`Private-ref`:  BB-8430

### CODE REVIEW

File: src/course-outline/configure-modal/ConfigureModal.jsx [hunk 1]
```diff
@@ -177,65 +246,55 @@ const ConfigureModal = ({
...
-          <ModalDialog.CloseButton variant="tertiary">
-            {intl.formatMessage(messages.cancelButton)}
-          </ModalDialog.CloseButton>
-          <Button onClick={handleSave} disabled={saveButtonDisabled}>
-            {intl.formatMessage(messages.saveButton)}
-          </Button>
-        </ActionRow>
-      </ModalDialog.Footer>
-    </ModalDialog>
+    isOpen && (
```
Comment (navinkarkera) [inline src/course-outline/configure-modal/ConfigureModal.jsx:269]: @CefBoud Is this check require?

File: src/course-outline/configure-modal/ConfigureModal.test.jsx [hunk 2]
```diff
@@ -283,3 +283,137 @@ describe('<ConfigureModal />', () => {
...
+      <ConfigureModal
+        isOpen
+        onClose={onCloseMock}
+        onConfigureSubmit={onConfigureSubmitMock}
+      />
+    </IntlProvider>,
+  </AppProvider>,
+);
+
+describe('<ConfigureModal />', () => {
```
Comment (navinkarkera) [inline src/course-outline/configure-modal/ConfigureModal.test.jsx]: Can you please update the description here and other describe blocks.

**Review by navinkarkera — APPROVED** [review]
@CefBoud Looks good. Have some minor nits.

:+1:
- [x] I tested this: (Tried out all options in unit configure modal)
- [x] I read through the code
- [ ] ~I checked for accessibility issues~
- [ ] ~Includes documentation~

File: src/course-outline/configure-modal/ConfigureModal.jsx (hunk 1 above)
Comment (CefBoud) [reply src/course-outline/configure-modal/ConfigureModal.jsx:269]: @navinkarkera I had some Undefined Props warnings on `displayName` and `category` IIRC.  This happens when there is no `currentItem`. The check for `isOpen` can be replaced with check(s) for `currenttItem`.

File: src/course-outline/configure-modal/ConfigureModal.test.jsx (hunk 2 above)
Comment (CefBoud) [reply src/course-outline/configure-modal/ConfigureModal.test.jsx]: Done!

**Comment by navinkarkera** [comment]
Will be merged as part of https://github.com/openedx/frontend-app-course-authoring/pull/797

**Pull request closed without merging**